chainntnfs/txnotifier.go

```diff
@@ -357,9 +357,28 @@ func NewSpendRequest(op *wire.OutPoint, pkScript []byte) (SpendRequest, error) {
 // String returns the string representation of the SpendRequest.
 func (r SpendRequest) String() string {
 	if r.OutPoint != ZeroOutPoint {
+		// If the pk script is all zeros, we blank the pk script.
+		// Currently we do not support taproot pk scripts for
+		// notifications.
+		if r.PkScript == ZeroTaprootPkScript {
+			return fmt.Sprintf("outpoint=%v, script=<zero> "+
+				"(taproot pk script not supported)",
+				r.OutPoint)
+		}
+
 		return fmt.Sprintf("outpoint=%v, script=%v", r.OutPoint,
 			r.PkScript)
 	}
+
+	// If the pk script is all zeros, we blank the pk script. Currently we
+	// do not support taproot pk scripts for notifications. This should
+	// never happen we should always at least have the outpoint for the
+	// taproot spends.
+	if r.PkScript == ZeroTaprootPkScript {
+		return fmt.Sprintf("outpoint=<zero>, script=<zero> " +
+			"(taproot pk script not supported)")
+	}
+
 	return fmt.Sprintf("outpoint=<zero>, script=%v", r.PkScript)
 }
```

Comment:
Nit: 
Could potentially do something like this instead, to not have to repeat the check both in and outside of the `r.OutPoint != ZeroOutPoint` case.   
```suggestion
func (r SpendRequest) String() string {
	outpointStr := fmt.Sprintf("%v", r.OutPoint)
	if r.OutPoint == ZeroOutPoint {
		outpointStr = "<zero>"
	}

	// If the pk script is all zeros, we blank the pk script.
	// Currently we do not support taproot pk scripts for notifications.
	// This should never happen when the output equals the ZeroOutPoint, a
	// we should always at least have the outpoint for the taproot spends.
	if r.PkScript == ZeroTaprootPkScript {
		return fmt.Sprintf("outpoint=%s, script=<zero> "+
			"(taproot pk script not supported)", outpointStr)
	}

	if r.OutPoint != ZeroOutPoint {
		return fmt.Sprintf("outpoint=%s, script=%v", outpointStr,
			r.PkScript)
	}

	return fmt.Sprintf("outpoint=<zero>, script=%v", r.PkScript)
}
```

Reply:
adopted your change, thank you!